app/modules/asset_groups/models.py

```diff
@@ -498,6 +498,202 @@ def get_all_jobs_debug(cls, verbose):
+            'numAttempts': self.detection_attempts,
+            'numAttemptsMax': MAX_DETECTION_ATTEMPTS,
+            'numAssets': None,
+            'numAnnotations': None,
+            'steps': 0,
+            'stepsComplete': 0,
+            'progress': None,
+        }
+        # TODO i am not sure if this is the best count here, as there
+        #   is config['updatedAssets'] which may actually need to be considered
+        status['numAssets'] = len(self.get_assets())
+        status['numAnnotations'] = len(self.get_all_annotations())
+
+        #  self.stage == AssetGroupSightingStage.detection
+        asset_group_config = self.asset_group.config
+        if 'speciesDetectionModel' in asset_group_config:
+            status['numModels'] = len(asset_group_config['speciesDetectionModel'])
+        # whose idea was this?
+        if (
+            status['numModels'] == 1
+            and asset_group_config['speciesDetectionModel'][0] == 'None'
+        ):
+            status['numModels'] = 0
+        if status['numModels'] < 1:  # assumed detection skipped
+            status['skipped'] = True
+            status['complete'] = True
+            status['steps'] = 1
+            status['stepsComplete'] = 1
+            return status
+
+        status['inProgress'] = True
+
+        if self.detection_attempts > MAX_DETECTION_ATTEMPTS:
+            status['error'] = f'could not start after {MAX_DETECTION_ATTEMPTS} attempts'
+            status['failed'] = True
+            status['inProgress'] = False
+            status['steps'] = 1
+
+        # TODO needs more exploration - can we find out if we are waiting to start in celery?
+        if not self.jobs:
+            status['numJobs'] = 0
+            status['numJobsActive'] = 0
+            status['numJobsFailed'] = 0
+            status['_debug'] = 'self.jobs is empty'
+            return status
+
+        # we reset failed here, as it looks like job started anyway?
+        status['steps'] = 1
+        status['stepsComplete'] = 1
+        status['error'] = None
+        status['failed'] = False
+        status['numJobs'] = len(self.jobs)
+        status['numJobsActive'] = 0
+        status['numJobsFailed'] = 0
+
+        job_info_list = []
+        for job_id in self.jobs:
+            job_info = {
+                'id': job_id,
+                'active': False,
+                'error': None,
+                'failed': False,
+                # we only query sage for *active* jobs to save expense,
+                #   so this being None is normal for inactive jobs
+                'sage_status': None,
+            }
+            job = self.jobs[job_id]
+            job_info['start'] = job.get('start')
+            job_info['end'] = job.get('end')
+            if job.get('active') is True:
+                job_info['active'] = True
+                status['numJobsActive'] += 1
+                # we sh/could toggle active if this shows failure  TODO
+                ss = None
```

Comment:
I dislike uber shortened names, it usually takes a fair bit of guesswork to figure out what they mean. 
Can you give this a real name please?

Reply:
will update this bad `ss` on next iteration of PR for this ticket (soon).   dont want to hold this up today for just this.